docs/enhancements/arm64-cpu-architecture.md

```diff
@@ -0,0 +1,149 @@
+---
+title: arm64-cpu-architecture
+authors:
+  - "@danielerez"
+creation-date: 2021-07-29
+last-updated: 2021-07-29
+---
+
+# ARM64 CPU architecture support
+
+## Summary
+
+The default host CPU architecture supported for installation is x86_64 in Assisted Installer. In order to support additional CPU architectures, the assisted-service should accept and handle selection of arm64 CPU architecture as well.
+
+## Motivation
+
+Although x86_64 is popular in many use cases, other CPU architectures are common in other segments.  For example, ARM is popular in edge use cases.  OpenShift has recently added support for arm64 and it is important to add this support to Assisted Installer.
+
+### Goals
+
+* Allow installation of clusters that are running on arm64 hosts.
+* The introduced infrastructure should facilitate future support of more architectures (e.g. PPC).
+
+### Non-Goals
+
+* The assisted-service will remain for x86_64 only.
+* No support for clusters with hosts of multiple architectures (e.g. x86_64 + arm64).
+* Implementation for kube-api (though design should take into account).
+* Day2 and late-binding won't be supporting in first phase.
+
+## Proposal
+
+### Host images
+
+The following images should be built also to arm64:
+  * assisted-installer
+  * assisted-installer-controller
+  * assisted-installer-agent
+
+This should be handled in similar to the current flows by modifying relevant Dockerfiles to support arm64 build. The images should be published to the same registries and follow same naming conventions. I.e. The best approach would be using the same image URI for any architecture. Which means, minimal changes in the assisted-service, as the architecture specific image would be fetched according to the machine pulling the image.
+
+### OCP versions
+
+[Versions](https://github.com/openshift/assisted-service/blob/master/internal/versions/versions.go) should
+have CPU architecture granularity for each API. So that OCP release images and RHCOS images of multiple architectures could be supported.
+
+The assisted-service should support the arm64 variant of RHCOS images and OCP release image (for OpenShift >= 4.9).
+
+The OPENSHIFT_VERSIONS environment variable should include Release/RHCOS images of multiple architectures for versions >= 4.9.
```

Comment:
Is backwards compatibility going to be guaranteed? 

Is it expected that users specifying a `4.9` image will have to use only the new format? Anyway to automatically convert from the old version of this variable to the new one?

Just mentioning things that may be worth noting in this enhancement.

Reply:
So the old format will still be supported (for backwards compatibility indeed). But, in order to use multiple architectures the proper images would have to be specified using the new format. So it is expected to use the new format if additional architectures are needed.

Reply:
If we're breaking compatibility, this may be the right opportunity to separate the definition of openshift release images from RHCOS images, and specify each separately as the kube API does. Perhaps `OPENSHIFT_VERSIONS` and `RHCOS_VERSIONS`. That would eliminate a lot of confusion and duplication. @avishayt WDYT?

Reply:
@mhrivnak Can you provide an example of `OPENSHIFT_VERSIONS` and `RHCOS_VERSIONS`?  What would this break for existing customers?

Reply:
Something like what's below. The enhancement proposes maintaining compatibility with the old OPENSHIFT_VERSIONS format in parallel with a new format. I'm just adjusting that new format here to put RHCOS image data elsewhere. The new format is backward-incompatible, but the key is to support both formats concurrently.

OS_IMAGES
```
"4.9": {
      "x86_64": {
            "url": "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest-4.9/rhcos-live.x86_64.iso",
            "rootfsurl": "https://mirror.openshift.com/pub/openshift-v4/x86_64/dependencies/rhcos/pre-release/latest-4.9/rhcos-live-rootfs.x86_64.img",
            "version": "49.84.202107032011-0"
      },
}
```

OPENSHIFT_VERSIONS
```
"4.9": {
   "display_name": "4.9.0-0.nightly",
   "support_level": "beta",
   "version": "4.9.0-0.nightly-2021-07-05-092650",
   "image": "quay.io/openshift-release-dev/ocp-release-nightly:4.9.0-0.nightly-2021-07-05-092650",
}
```